drivers/modem/Kconfig.wncm14a2a

```diff
@@ -10,6 +10,7 @@ menuconfig MODEM_WNCM14A2A
 	bool "Enable Wistron LTE-M modem driver"
 	select MODEM_RECEIVER
 	select NET_OFFLOAD
+	select GPIO
```

Comment:
(Maybe this could be a `depends on` as well, but `GPIO` has no dependencies at least.)

Reply:
@ulfalizer Ugh, I forget: does using select this way *not* include the dependencies for GPIO (if it had any?).  I would prefer to have the driver select dependencies if it needs them, rather than placing that logic on the samples or boards that select the driver.

Reply:
`select` forces symbols on, ignoring dependencies. It's risky for that reason.

I wrote up a section on `select` pitfalls and recommendations on the [Kconfig - Tips and Best Practices](https://docs.zephyrproject.org/latest/guides/kconfig/index.html#select-statements) page.

Reply:
@ulfalizer Right.  I remember that.  So, `select` has a very narrow use-case where it's only safe to use it when it's in a small set of options that are all known good configurations.  In place of that, we're now adding `depends on` throughout the  Kconfig options and deferring the logic needed to enable drivers into the HW and/or the sample configurations.   And, there was a push to remove the defaults from the HW configurations recently, which leaves only the samples.

... This doesn't seem like it's going end well.

Doesn't the Linux kernel have a mechanism to accomplish what we are doing?

Reply:
Could we introduce a PR test which checks to see if configs which are `select`ed have dependencies and if so: rejects the change?  There should be some basic configs which drivers can select (without fear of future breakage): ADC, AUDIO, GPIO, I2C, PWM, etc ...

Reply:
I wonder if such a check might be too broad.

For example, say you have a bunch of symbols for audio codecs that are all defined next to each other, and that all have the same (say complex) dependencies. In that case, it might not be horrible to have a symbol like `CONFIG_ALL_FOO_CODECS` next to them that selects all of the FOO-related ones.

There are some legitimate and reasonably safe uses for `select` like that. It's mostly when people start treating it as "just another `depends on` alternative" that things tend to get messy real quick. :)

Check out the [imply](https://docs.zephyrproject.org/latest/guides/kconfig/index.html#the-imply-statement) statement too. It enables symbols only if their dependencies are satisfied.

Reply:
I'll switch to `imply` usage here and repost.